graphrag/index/graph/extractors/community_reports/sort_context.py

```diff
@@ -154,3 +154,81 @@ def _get_context_string(
         )
 
     return context_string
+
+
+def sort_context_batch(
+    local_contexts: pd.DataFrame,
+    node_details_column: str = schemas.NODE_DETAILS,
+    edge_details_column: str = schemas.EDGE_DETAILS,
+    claim_details_column: str = schemas.CLAIM_DETAILS,
+    community_id_column: str = schemas.COMMUNITY_ID,
+    max_tokens: int | None = None,
+) -> pd.DataFrame:
+    """Batch processing for community context strings."""
+
+    def generate_context(group):
+        # Explode and deduplicate edges and claims
+        edges = pd.DataFrame(
+            group[edge_details_column].explode().dropna().tolist()
+        ).drop_duplicates()
+        claims = pd.DataFrame(
+            group[claim_details_column].dropna().tolist()
+        ).drop_duplicates()
+        nodes = pd.DataFrame(
+            group[node_details_column].dropna().tolist()
+        ).drop_duplicates()
+
+        # Sort dataframes for consistent output
+        if not nodes.empty:
+            nodes.sort_values(
+                by=[schemas.NODE_DEGREE, schemas.NODE_ID],
+                ascending=[False, True],
+                inplace=True,
+            )
+        if not edges.empty:
+            edges.sort_values(
+                by=[schemas.EDGE_DEGREE, schemas.EDGE_ID],
+                ascending=[False, True],
+                inplace=True,
+            )
+        if not claims.empty:
+            claims.sort_values(by=schemas.CLAIM_ID, ascending=True, inplace=True)
+
+        # Generate context string
+        contexts = []
+        if not nodes.empty and len(nodes) > 0:
```

Comment:
is it possible for `not nodes.empty` to be True and `len(nodes)` = 0?

Reply:
yup, empty checks for shape, so either having columns or rows works, and, in this case, we end up with columns. Maybe just checking len will be enoug

Reply:
Dunno, let me check